src/main/java/org/gridsuite/securityanalysis/server/repositories/specifications/AbstractCommonSpecificationBuilder.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.securityanalysis.server.repositories.specifications;
+
+import org.gridsuite.securityanalysis.server.dto.ResourceFilterDTO;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.List;
+import java.util.UUID;
+/**
+ * @author Kevin LE SAULNIER <kevin.lesaulnier@rte-france.com>
+ */
+
+public abstract class AbstractCommonSpecificationBuilder<T> {
+    AbstractCommonSpecificationBuilder() {
+    }
+
+    public Specification<T> resultUuidEquals(UUID value) {
+        return (contingency, cq, cb) -> cb.equal(contingency.get("result").get("id"), value);
```

Comment:
It will be more flexible and readble to: first rename contingency by root , and not harcode the fields names and do something like this :
public abstract String getResultFieldName();
public abstract String getIdFieldName();
public Specification<T> resultUuidEquals(UUID value) {
    return (root, cq, cb) -> cb.equal(root.get(getResultFieldName()).get(getIdFieldName()), value);
}
and implement this two methods in the subclasses

Reply:
fixed with somethings similar